crates/forge_main/src/model.rs

```diff
@@ -31,6 +32,125 @@ impl From<&[Model]> for Info {
     }
 }
 
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct ForgeCommand {
+    pub name: String,
+    pub description: String,
+}
+
+impl From<HashMap<String, String>> for ForgeCommandManager {
+    fn from(value: HashMap<String, String>) -> Self {
+        ForgeCommandManager::default().register_all(value.into_iter().map(
+            |(command, description)| {
+                let name = if command.starts_with("/") {
+                    command
+                } else {
+                    format!("/{}", command)
+                };
+                ForgeCommand { name, description }
+            },
+        ))
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct ForgeCommandManager {
+    commands: Vec<ForgeCommand>,
+}
+
+impl Default for ForgeCommandManager {
+    fn default() -> Self {
+        ForgeCommandManager { commands: vec![] }
```

Comment:
Can we use strum to iterate over all the commands and insert it. May be create a `From<Command> for ForgeCommand` implementation to make it simpler to convert.